webclient/api/src/main/java/io/helidon/webclient/api/Proxy.java

```diff
@@ -469,26 +472,26 @@ private static Socket connectToProxy(WebClient webClient,
                                                                     it -> {
                                                                     })
                 .connect();
-
-        HttpClientRequest request = webClient.method(Method.CONNECT)
-                .followRedirects(false) // do not follow redirects for proxy connect itself
-                .connection(connection)
-                .uri("http://" + proxyAddress.getHostName() + ":" + proxyAddress.getPort())
-                .protocolId("http/1.1") // MUST be 1.1, if not available, proxy connection will fail
-                .header(HeaderNames.HOST, targetAddress.getHostName() + ":" + targetAddress.getPort())
-                .accept(MediaTypes.WILDCARD);
-        if (clientConfig.keepAlive()) {
-            request.header(HeaderValues.CONNECTION_KEEP_ALIVE)
-                .header(PROXY_CONNECTION);
-        }
-        proxy.proxyAuthHeader.ifPresent(request::header);
-        // we cannot close the response, as that would close the connection
-        HttpClientResponse response = request.request();
-        if (response.status().family() != Status.Family.SUCCESSFUL) {
-            response.close();
-            throw new IllegalStateException("Proxy sent wrong HTTP response code: " + response.status());
+        if (proxy.forceHttpConnect || tls || proxy.username.isPresent()) {
```

Comment:
Proxy-Connection header should be set even when not using CONNECT (at least that is what curl does, and I would think that that is correct)

Reply:
From the code we have, it seems currently we only set `Proxy-Connection` in the first request to the Proxy. This request may not happen with this new changes. Should we send always `Proxy-Connection` to every request when using proxy?.

From the spec, `Proxy-Connection` is a redundant header that has the same effect than `Connection`. So there is no point on setting it.

Reply:
Looks like curl removed `Proxy-Connection` based on specs as you mentioned, but put it back because of a slew of bugs filed against it when it was removed: https://stackoverflow.com/questions/62722430/why-does-curl-send-a-proxy-connection-header-even-though-the-rfc-seems-to-disco. Seems that there are still lots of proxy servers out there that it s expecting this header.

Reply:
Nice finding @klustria 

So, my understanding is that if proxy is set, we need to set `Proxy-Connection` in all the requests.

Reply:
Yeah, that's what I understand too.

Reply:
Done. It is excluded in HTTP2 as it is said here: https://www.rfc-editor.org/rfc/rfc9113.html

[8.2.2. ](https://www.rfc-editor.org/rfc/rfc9113.html#section-8.2.2)[Connection-Specific Header Fields](https://www.rfc-editor.org/rfc/rfc9113.html#name-connection-specific-header-)

HTTP/2 does not use the Connection header field ([Section 7.6.1](https://www.rfc-editor.org/rfc/rfc9110#section-7.6.1) of [[HTTP](https://www.rfc-editor.org/rfc/rfc9113.html#RFC9110)]) to indicate connection-specific header fields; in this protocol, connection-specific metadata is conveyed by other means. An endpoint MUST NOT generate an HTTP/2 message containing connection-specific header fields. This includes the Connection header field and those listed as having connection-specific semantics in [Section 7.6.1](https://www.rfc-editor.org/rfc/rfc9110#section-7.6.1) of [[HTTP](https://www.rfc-editor.org/rfc/rfc9113.html#RFC9110)] (that is, Proxy-Connection, Keep-Alive, Transfer-Encoding, and Upgrade). Any message containing connection-specific header fields MUST be treated as [malformed](https://www.rfc-editor.org/rfc/rfc9113.html#malformed) ([Section 8.1.1](https://www.rfc-editor.org/rfc/rfc9113.html#malformed)).
The only exception to this is the TE header field, which MAY be present in an HTTP/2 request; when it is, it MUST NOT contain any value other than "trailers".An intermediary transforming an HTTP/1.x message to HTTP/2 MUST remove connection-specific header fields as discussed in [Section 7.6.1](https://www.rfc-editor.org/rfc/rfc9110#section-7.6.1) of [[HTTP](https://www.rfc-editor.org/rfc/rfc9113.html#RFC9110)], or their messages will be treated by other HTTP/2 endpoints as [malformed](https://www.rfc-editor.org/rfc/rfc9113.html#malformed) ([Section 8.1.1](https://www.rfc-editor.org/rfc/rfc9113.html#malformed)).